homeassistant/components/modbus/validators.py

```diff
@@ -189,3 +191,35 @@ def scan_interval_validator(config: dict) -> dict:
             )
             hub[CONF_TIMEOUT] = minimum_scan_interval - 1
     return config
+
+
+def duplicate_entity_validator(config: dict) -> dict:
+    """Control scan_interval."""
+    for hub_index, hub in enumerate(config):
+        addresses = []
+        for component, conf_key in PLATFORMS:
+            if conf_key not in hub:
+                continue
+            names = []
+            errors = []
+            for index, entry in enumerate(hub[conf_key]):
+                name = entry[CONF_NAME]
+                addr = str(entry[CONF_ADDRESS])
+                if CONF_SLAVE in entry:
+                    addr += "_" + str(entry[CONF_SLAVE])
+                if addr in addresses:
+                    err = f"Modbus {component}/{name} address {addr} is duplicate, second entry not loaded!"
+                    _LOGGER.warning(err)
+                    errors.append(index)
+                elif name in names:
+                    err = f"Modbus {component}/{name}  is duplicate, second entry not loaded!"
+                    _LOGGER.warning(err)
+                    errors.append(index)
+                else:
+                    names.append(name)
+                    addresses.append(addr)
```

Comment:
```suggestion
        addresses = set()
        for component, conf_key in PLATFORMS:
            if conf_key not in hub:
                continue
            names = set()
            errors = []
            for index, entry in enumerate(hub[conf_key]):
                name = entry[CONF_NAME]
                addr = str(entry[CONF_ADDRESS])
                if CONF_SLAVE in entry:
                    addr += "_" + str(entry[CONF_SLAVE])
                if addr in addresses:
                    err = f"Modbus {component}/{name} address {addr} is duplicate, second entry not loaded!"
                    _LOGGER.warning(err)
                    errors.append(index)
                elif name in names:
                    err = f"Modbus {component}/{name}  is duplicate, second entry not loaded!"
                    _LOGGER.warning(err)
                    errors.append(index)
                else:
                    names.add(name)
                    addresses.add(addr)
```

Reply:
Thanks I learned something today (.add works, I had only looked for .append and .extend).